### PR TITLE
Add docs for Encoding, IREECodegen, and IREEGPU dialects.

### PR DESCRIPTION
**Pull request by ScottTodd**
These will be visible at https://iree.dev/reference/mlir-dialects/

Previews:

* https://scotttodd.github.io/iree/reference/mlir-dialects/Encoding/
* https://scotttodd.github.io/iree/reference/mlir-dialects/IREECodegen/
* https://scotttodd.github.io/iree/reference/mlir-dialects/IREEGPU/

### CODE REVIEW

File: docs/website/docs/reference/mlir-dialects/index.md [hunk 1]
```diff
@@ -19,11 +19,14 @@ these dialects can be found in the
...
 --------------------------- | -----------
 [Check](./Check.md)         | Defines assertions for IREE tests
+[Encoding](./Encoding.md)   | Tensor encoding attributes and related ops
 [Flow](./Flow.md)           | Models execution data flow and partitioning
 [HAL](./HAL.md)             | Represents operations against the IREE HAL[^1]
 [HAL/Inline](./HALInline.md) | Inline HAL interop runtime module dialect
 [HAL/Loader](./HALLoader.md) | HAL inline executable loader runtime module dialect
 [IO/Parameters](./IOParameters.md) | External parameter resource management APIs
+[IREECodegen](./IREECodegen.md) | Attributes used by IREE code generation
+[IREEGPU](./IREEGPU.md) | Attributes used by GPU focused IREE code generation
```
Comment (Groverkss) [inline docs/website/docs/reference/mlir-dialects/index.md]: I don't think it's just attributes. It's some operations as well

**Comment by andfau-amd** [comment]
What about VectorExt, should that also be there?

File: docs/website/docs/reference/mlir-dialects/index.md (hunk 1 above)
Comment (qedawkins) [reply docs/website/docs/reference/mlir-dialects/index.md]: IREECodegen also has an op for ukernels. Something like `Common functionality used by [GPU focused] IREE code generation`.

**Review by qedawkins — APPROVED** [review]

File: docs/website/docs/reference/mlir-dialects/index.md (hunk 1 above)
Comment (ScottTodd) [reply docs/website/docs/reference/mlir-dialects/index.md]: Thanks, updated the summaries.

**Pull request merged**